file-formats/aifi/type/zif_aff_aifi_v1.intf.abap

```diff
@@ -0,0 +1,496 @@
+      "! Category field name
+      category_fieldname          TYPE c LENGTH 32,
+      " /AIF/IS_HIDDEN, swap to positive stature
+      "! <p class="shorttext">Show Tree Node</p>
+      "! Show tree node in view 1 tree
+      "! $default 'X'
+      is_show_tree_node           TYPE abap_bool,
+      " (type /aif/alertrelevant)
+      "! <p class="shorttext">Relevant for Alert Recipient Determination</p>
+      "! Relevant for alert recipient determination
+      is_alert_recipient_relevant TYPE abap_bool,
+    END OF ty_key_field_selection.
+
+  TYPES:
+    "! <p class="shorttext">SAP Fiori Features</p>
+    "! SAP Fiori specific features
+    BEGIN OF ty_fiori_features,
+      " (type /aif/semantic_object)
+      "! <p class="shorttext">Semantic Object</p>
+      "! Semantic object
+      semantic_object            TYPE zif_aff_types_v1=>ty_object_name_30,
+      " (type /aif/semantic_action)
+      "! <p class="shorttext">Semantic Action</p>
+      "! Semantic action
+      semantic_action            TYPE zif_aff_types_v1=>ty_description_60,
+      " /AIF/NOT_SHOW_FIELD
+      "! <p class="shorttext">Display in Message Monitoring</p>
+      "! Display in message monitoring
+      "! $default 'X'
+      is_message_monitor_show    TYPE abap_bool,
+      " /AIF/IS_DEFAULT_FIELD
+      "! <p class="shorttext">Display in Message Monitoring by Default</p>
+      "! Display in message monitoring by default
+      is_message_monitor_default TYPE abap_bool,
+    END OF ty_fiori_features.
+
+  TYPES:
+    " Related key fields used by key field rules
+    "! <p class="shorttext">Rule Key Field Name</p>
+    "! Rule key field name
+    BEGIN OF ty_rule_key_field_name,
+      " (type /aif/key_fieldname_enh)
+      "! <p class="shorttext">Key Field Name</p>
+      "! Key field name
+      rule_key_field_name  TYPE zif_aff_types_v1=>ty_object_name_30,
+      " (type /aif/field_number_enh)
+      "! <p class="shorttext">Field Sequence Number</p>
+      "! Field sequence number
+      field_sequnce_number TYPE c LENGTH 3,
+    END OF ty_rule_key_field_name.
+
+  "! <p class="shorttext">Key Field Rules</p>
+  "! Key field rules
+  TYPES ty_rule_key_field_names TYPE STANDARD TABLE OF ty_rule_key_field_name WITH DEFAULT KEY.
+
+  TYPES:
+    "! <p class="shorttext">Key Field</p>
+    "! Key field
+    BEGIN OF ty_key_field,
+      "! <p class="shorttext">General</p>
+      "! General key field settings
+      key_field_settings   TYPE ty_key_field_settings,
+      "! <p class="shorttext">Key Field Determination by Qualifier</p>
+      "! Key field determination by qualifier
+      key_field_qualifier  TYPE ty_key_field_qualifier,
+      "! <p class="shorttext">Key Field Selection</p>
+      "! Key field selection type and settings
+      key_field_selection  TYPE ty_key_field_selection,
+      "! <p class="shorttext">Enable Key Field Rules</p>
+      "! Enable key field rules
+      rule_key_field_names TYPE ty_rule_key_field_names,
```

Comment:
Why do you have `Enable` in the title? Aren't these just `Key Field Rules`?

Reply:
corrected